src/main/java/lotto/domain/Prize.java

```diff
@@ -0,0 +1,66 @@
+package lotto.domain;
+
+public enum Prize {
+    NOTHING(0L, 2, 999 ),
+    FIFTH(5000L, 3, 5),
+    FOURTH(50000L, 4, 4),
+    THIRD(1500000L, 5, 3),
+    SECOND(30000000L, 5, 2),
+    FIRST(2000000000L, 6, 1);
+
+    private final Long reward;
+    private final int matchCount;
+    private final int order;
+
+    Prize(Long reward, int matchCount, int order) {
+        this.reward = reward;
+        this.matchCount = matchCount;
+        this.order = order;
+    }
+
+    public Long totalReward(Long count) {
+        return this.reward * count;
+    }
+
+    public int getOrder() {
+        return order;
+    }
+
+    public static Prize evaluate(int matchCount, boolean isBonusNumberMatched) {
+        if (matchCount <= 2) {
+            return NOTHING;
+        }
+        if (matchCount == 3) {
+            return FIFTH;
+        }
+        if (matchCount == 4) {
+            return FOURTH;
+        }
+        if (matchCount == 5) {
+            return evaluate(isBonusNumberMatched);
+        }
+        return FIRST;
+    }
+
+    private static Prize evaluate(boolean isBonusNumberMatched) {
+        if (isBonusNumberMatched) {
+            return SECOND;
+        }
+        return THIRD;
+    }
```

Comment:
지금과 같이 처리하게되면 Prize가 하나 추가되게되면 또 분기문이 추가될거에요. 🥲  (서비스가 커짐에따라서 이 분기문은 계속 늘어나겠죠...? 😱)

matchoCount가 5인 경우를 제외하면 아래와 같이 변경해볼 수 있을 것 같아요.
이렇게 작성하면 (5 케이스와 같은 예외적인 케이스를 제외하고) 일반적인 케이스의 요소가 추가되더라도 분기문을 추가할 필요가 없지 않을까 싶어요. 😉

```
```suggestion
    public static Prize evaluate(int matchCount, boolean isBonusNumberMatched) {
        // 5 케이스 처리 
        
        return Arrays.stream(values())
            .filter(prize -> prize.matchCount == matchCount)
            .findAny()
            .orElse(NOTHING);
    }
```